### PR TITLE
Misc/validate collections documents

### PR DESCRIPTION
**Pull request by csadorf**

### CODE REVIEW

File: signac/contrib/collection.py [hunk 1]
```diff
@@ -493,6 +494,24 @@ def __getitem__(self, _id):
             self._assert_open()
             raise
 
+    @staticmethod
+    def _validate_key(key):
+        "Emit a warning or raise an exception if key is invalid. Returns key."
+        if '.' in key:
+            from ..errors import InvalidKeyError
```
Comment (vyasr) [inline signac/contrib/collection.py]: We normally don't put imports nested anywhere unless it's a `try...except` block, is there a reason not to put this at the top of the file?

File: signac/contrib/collection.py [hunk 2]
```diff
@@ -503,7 +522,7 @@ def __setitem__(self, _id, doc, _trust=False):
         if _trust:
             self._docs[_id] = doc
         else:
-            self._docs[_id] = json.loads(json.dumps(doc))
+            self._docs[_id] = self._validate_doc(json.loads(json.dumps(doc)))
```
Comment (vyasr) [inline signac/contrib/collection.py]: Is the `loads(dumps(doc))` just to ensure that the document is valid JSON? If not, then this is a no-op, right? And if so, should we wrap it in a `try...except` to provide a more informative error?

File: signac/contrib/collection.py (hunk 1 above)
Comment (csadorf) [reply signac/contrib/collection.py]: Yes, because there are otherwise module interdependencies.

File: signac/contrib/collection.py (hunk 2 above)
Comment (csadorf) [reply signac/contrib/collection.py]: Yes, exactly. I'll see if I can easily provide an informative error message.

Comment (csadorf) [reply signac/contrib/collection.py]: I forgot to mention, it also ensures type normalization.

**Pull request merged**